File: fingerprint/src/main/java/com/simprints/fingerprint/activities/collect/FingerFragment.java
package com.simprints.fingerprint.activities.collect;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.simprints.fingerprint.R;
import com.simprints.fingerprint.activities.collect.models.Finger;
import com.simprints.fingerprint.activities.collect.models.FingerRes;
import com.simprints.fingerprint.activities.collect.models.FingerStatus;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

public class FingerFragment extends Fragment {

    private final static String FINGER_ARG = "finger";
    private TextView resultText;
    private TextView fingerNumber;
    private TextView directionText;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_finger, container, false);
        resultText = view.findViewById(R.id.finger_result_text);
        fingerNumber = view.findViewById(R.id.finger_number_text);
        directionText = view.findViewById(R.id.finger_direction_text);

        Finger finger = (Finger) getArguments().get(FINGER_ARG);
        assert finger != null;

        FingerRes.setFingerRes();

        ImageView fingerImageView = view.findViewById(R.id.finger_image);
        fingerImageView.setImageResource(FingerRes.get(finger).getDrawableId());
        fingerImageView.setVisibility(View.VISIBLE);

        updateTextAccordingToStatus();

        return view;
    }

    public void updateTextAccordingToStatus() {
        Finger finger = (Finger) getArguments().get(FINGER_ARG);
        assert finger != null;

        FingerRes fingerRes = FingerRes.get(finger);

        resultText.setText(finger.getStatus().getTextResult());
        resultText.setTextColor(getResources().getColor(finger.getStatus().getTextResultColorRes()));
        fingerNumber.setText(getString(fingerRes.getNameId()));
        fingerNumber.setTextColor(getResources().getColor(R.color.simprints_blue));
        if (finger.isLastFinger() && (finger.getStatus() == FingerStatus.GOOD_SCAN
            || finger.getStatus() == FingerStatus.RESCAN_GOOD_SCAN)) {
            directionText.setText(R.string.empty);
        } else {
            directionText.setText(finger.getStatus().getTextDirection());
        }
        directionText.setTextColor(finger.getStatus().getTextDirectionColor());
    }

    public static FingerFragment newInstance(Finger finger) {
        FingerFragment fingerFragment = new FingerFragment();
        Bundle bundle = new Bundle();
        bundle.putParcelable(FINGER_ARG, finger);
        fingerFragment.setArguments(bundle);
        return fingerFragment;
    }

    public Finger getFinger() {
        Finger finger = getArguments().getParcelable(FINGER_ARG);
        assert finger != null;
        return finger;
    }
}

File: fingerprint/src/main/java/com/simprints/fingerprint/activities/collect/FingerFragment.kt
package com.simprints.fingerprint.activities.collect

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat.getColor
import androidx.fragment.app.Fragment
import com.simprints.fingerprint.R
import com.simprints.fingerprint.activities.collect.models.Finger
import com.simprints.fingerprint.activities.collect.models.FingerRes
import com.simprints.fingerprint.activities.collect.models.FingerStatus
import com.simprints.fingerprint.controllers.core.androidResources.FingerprintAndroidResourcesHelper
import com.simprints.fingerprint.controllers.core.preferencesManager.FingerprintPreferencesManager
import com.simprints.fingerprint.tools.extensions.activityIsPresentAndFragmentIsAdded

class FingerFragment : Fragment() {

    lateinit var androidResourcesHelper: FingerprintAndroidResourcesHelper
    lateinit var fingerprintPreferencesManager: FingerprintPreferencesManager

    lateinit var finger: Finger
    private lateinit var fingerImage: ImageView
    private lateinit var fingerResultText: TextView
    private lateinit var fingerDirectionText: TextView
    private lateinit var fingerNumberText: TextView

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_finger, container, false)

        finger = arguments?.get(FINGER_ARG) as Finger

        FingerRes.setFingerRes()

        fingerImage = view.findViewById(R.id.fingerImage)
        fingerResultText = view.findViewById(R.id.fingerResultText)
        fingerDirectionText = view.findViewById(R.id.fingerDirectionText)
        fingerNumberText = view.findViewById(R.id.fingerNumberText)

        if (activityIsPresentAndFragmentIsAdded()) {
            updateOrHideFingerImageAccordingToSettings()
            updateTextAccordingToStatus()
        }

        return view
    }

    private fun updateOrHideFingerImageAccordingToSettings() {
        if (fingerprintPreferencesManager.fingerImagesExist) {
            updateFingerImageAccordingToStatus()
        } else {
            fingerImage.visibility = View.INVISIBLE
        }
    }

    private fun updateFingerImageAccordingToStatus() {
        fingerImage.setImageResource(FingerRes.get(finger).drawableId)
        fingerImage.visibility = View.VISIBLE
    }

    fun updateTextAccordingToStatus() {
        updateFingerResultText()
        updateFingerNumberText()
        updateFingerDirectionText()
    }

    private fun updateFingerResultText() {
        fingerResultText.text = androidResourcesHelper.getString(finger.status.textResult)
        fingerResultText.setTextColor(getColor(requireContext(), finger.status.textResultColorRes))
    }

    private fun updateFingerNumberText() {
        fingerNumberText.text = androidResourcesHelper.getString(FingerRes.get(finger).nameId)
        fingerNumberText.setTextColor(getColor(requireContext(), R.color.simprints_blue))
    }

    private fun updateFingerDirectionText() {
        if (finger.isLastFinger &&
            (finger.status == FingerStatus.GOOD_SCAN ||
                finger.status == FingerStatus.RESCAN_GOOD_SCAN)) {
            fingerDirectionText.text = androidResourcesHelper.getString(R.string.empty)
        } else {
            fingerDirectionText.text = androidResourcesHelper.getString(finger.status.textDirection)
        }
        fingerDirectionText.setTextColor(finger.status.textDirectionColor)
    }

    companion object {

        private const val FINGER_ARG = "finger"

        fun newInstance(
            finger: Finger,
            androidResourcesHelper: FingerprintAndroidResourcesHelper,
            fingerprintPreferencesManager: FingerprintPreferencesManager
        ): FingerFragment {
            val fingerFragment = FingerFragment()
            val bundle = Bundle()
            bundle.putParcelable(FINGER_ARG, finger)
            fingerFragment.arguments = bundle
            fingerFragment.androidResourcesHelper = androidResourcesHelper
            fingerFragment.fingerprintPreferencesManager = fingerprintPreferencesManager
            return fingerFragment
        }
    }
}
